ingestion/src/kldata.rs

```diff
@@ -35,47 +44,84 @@ struct ObsinnId {
     sensor_and_level: Option<(i32, i32)>,
 }
 
-fn parse_meta_field<T: FromStr>(field: &str, expected_key: &'static str) -> Result<T, Error>
-where
-    <T as FromStr>::Err: Debug,
-{
-    let (key, value) = field.split_once('=').ok_or_else(|| {
-        Error::Parse(format!(
-            "unexpected field in kldata header format: {}",
-            field
-        ))
-    })?;
+// TODO: maybe this can be a field in ObsinnChunk?
+#[derive(Default)]
+struct ObsinnHeader {
+    station_id: Option<i32>,
+    type_id: Option<i32>,
+    message_id: usize,
+    // TODO: we can parse it to Datatime if we decide we are going to use it
+    received_time: Option<String>,
+}
 
-    if key != expected_key {
-        return Err(Error::Parse(format!(
-            "wrong key in field: expected '{}', got '{}'",
-            expected_key, key
-        )));
-    }
+impl ObsinnHeader {
+    fn parse(meta: &str) -> Result<Self, Error> {
+        let mut fields = meta.split('/');
 
-    let res = value
-        .parse::<T>()
-        .map_err(|_| Error::Parse(format!("invalid number in kldata header for key {}", key)))?;
+        let kldata_string = fields
+            .next()
+            .ok_or_else(|| Error::Parse("kldata header terminated early".to_string()))?;
 
-    Ok(res)
-}
+        if kldata_string != "kldata" {
+            return Err(Error::Parse(
+                "kldata indicator missing or out of order".to_string(),
+            ));
+        }
 
-fn parse_meta(meta: &str) -> Result<(i32, i32, usize), Error> {
-    let mut parts = meta.splitn(4, '/');
+        let unexpected_field = |field: &str| {
+            Error::Parse(format!(
+                "unexpected field in kldata header format: {}",
+                field
+            ))
+        };
 
-    let next_err = || Error::Parse("kldata header terminated early".to_string());
+        let mut header = ObsinnHeader::default();
 
-    if parts.next().ok_or_else(next_err)? != "kldata" {
-        return Err(Error::Parse(
-            "kldata indicator missing or out of order".to_string(),
-        ));
-    }
+        for field in fields.by_ref() {
+            // TODO: this field signals data deletion/update in kvalobs, we do not use it
```

Comment:
??? Where did you find out about this from? How does it work? I thought our conversations with Søren concluded that messages are implicitly deletions/updates if that (station_id, timestamp) had been sent before?

Reply:
From a conversation with Søren that I should have probably recorded :rofl: I seem to recall that when the `add` keyword is specified the values for the different timestamps are updated only for the given `param_codes` and deleted for the others, but that doesn't sound quite right, I'll try to ask him again :facepalm: